sovrin_client/test/cli/test_agent_cli.py

```diff
@@ -0,0 +1,128 @@
+    connectIfNotAlreadyConnected
+
+
+@Spyable(methods=[AgentCli.print, AgentCli.printTokens])
+class TestAgentCLI(AgentCli, TestCliCore):
+    pass
+
+
+@pytest.fixture(scope='module')
+def agentCliBuilder(tdir, tdirWithPoolTxns, tdirWithDomainTxns, tconf, cliTempLogger):
+    return partial(getCliBuilder, tdir=tdir, tconf=tconf,
+                   tdirWithPoolTxns=tdirWithPoolTxns,
+                   tdirWithDomainTxns=tdirWithDomainTxns,
+                   logFileName=cliTempLogger, cliClass=TestAgentCLI)
+
+
+@pytest.fixture(scope='module')
+def acmeAgentCli(agentCliBuilder, acmeAgentPort):
+    yield from agentCliBuilder(
+        name='Acme-Agent',
+        agentCreator=lambda: createAcme(port=acmeAgentPort)
+    )('Acme-Agent')
+
+
+@pytest.fixture(scope='module')
+def acmeAgentCliRunning(acmeWithEndpointAdded, acmeAgentCli, looper):
+    looper.run(acmeAgentCli.agent.bootstrap())
+    return acmeAgentCli
+
+
+def test_acme_cli_started_successfully(be, acmeAgentCliRunning):
+    be(acmeAgentCliRunning)
+    assert acmeAgentCliRunning.currPromptText == 'Acme-Agent'
+
+
+def sendProofRequest(be, do, agentCli, userMap):
+    be(agentCli)
+    do('send proof-request {pr-name} to {send-proof-target}',
+       within=5,
+       mapper=userMap,
+       expect=[
+           'Sent proof request "{pr-name}" to {send-proof-target}'
+       ])
+
+
+def checkProofRequestReceived(be, do, userCli, commandMap):
+    be(userCli)
+    do(None, within=3, mapper=commandMap,
+       expect=['Proof request {pr-name} received from {inviter}.'])
+
+
+def getProofRequestsCount(userCli, target):
+    li = userCli.activeWallet.getLinkInvitationByTarget(target)
+    return len(li.proofRequests)
+
+
+@pytest.fixture(scope='module')
+def aliceAcceptedAcmeInvitationNoProofReq(
+        acmeAgentCliRunning, be, do, aliceCLI, acmeMap, loadInviteOut,
+        unsycedAcceptedInviteWithoutClaimOut, connectedToTest,
+        syncLinkOutWithEndpoint, newKeyringOut):
+    def _(invitationFile, keyring):
+        be(aliceCLI)
+        connectIfNotAlreadyConnected(do, connectedToTest, aliceCLI, acmeMap)
+        keyringMapper = {
+            'keyring-name': keyring
+        }
+        do('new keyring {}'.format(keyring),
+           expect=newKeyringOut,
+           mapper=keyringMapper)
+        do('load {}'.format(invitationFile),
+           mapper=acmeMap,
+           expect=loadInviteOut)
+        do('sync {inviter}',
+           mapper=acmeMap,
+           expect=syncLinkOutWithEndpoint,
+           within=15)
+        do('accept invitation from {inviter}',
+           within=15,
+           mapper=acmeMap,
+           expect=unsycedAcceptedInviteWithoutClaimOut)
+
+        proofRequestsBefore = getProofRequestsCount(aliceCLI, acmeMap['target'])
+
+        sendProofRequest(be, do, acmeAgentCliRunning, acmeMap)
+
+        checkProofRequestReceived(be, do, aliceCLI, acmeMap)
+
+        return proofRequestsBefore
```

Comment:
Since receiving proof request is async operation and we need to check whether count of before and after are same or incremented by 1. That operation is performed in `eventually`, that's why we just return before value and after value is matched in test which is using this fixture. Assert conditions for both of them are different.